src/Storages/StorageFile.cpp

```diff
@@ -208,7 +210,7 @@ std::unique_ptr<ReadBuffer> createReadBuffer(
             throwFromErrno("Cannot stat file " + current_path, ErrorCodes::CANNOT_STAT);
 
         if (S_ISREG(file_stat.st_mode))
-            nested_buffer = std::make_unique<ReadBufferFromFilePRead>(current_path, context->getSettingsRef().max_read_buffer_size);
+            nested_buffer = std::make_unique<MMapReadBufferFromFile>(current_path, 0);
```

Comment:
This currently doesn't respect the `max_read_buffer_size` settings, leading to several test failures. Will think about how to fix it tomorrow.

Reply:
It is ok - the setting `max_read_buffer_size` is irrelevant for the case when `mmap` is used for reading.

How to adjust the tests:
- introduce a setting `storage_file_allow_mmap` with value 1 by default and set it to 0 in those tests.

It makes sense to introduce this setting nevertheless as a feature flag (to allow users to turn the new behavior on and off without downgrading in case of any troubles happen).

Reply:
In the end, I've decided completely move all this logic to already existing functionality in `Disks/IO/createReadBufferFromFileBase`. I also thought to add a new setting but found the `local_filesystem_read_method` settings, which is basically doing what we are looking for. It also makes sense that `StorageFile` itself should respect `IO/ReadSettings`.